src/sage/rings/lazy_series_ring.py

```diff
@@ -1792,6 +1793,43 @@ def residue_field(self):
             raise TypeError("the base ring is not a field")
         return R
 
+    def taylor(self, f):
+        r"""
+        Return the Taylor expansion around `0` of the function ``f``.
+
+        INPUT:
+
+        - ``f`` -- a function such that one of the following works:
+
+          * the substitution `f(z)`, where `z` is generator of ``self``
+          * `f` is a function of a single variable with no poles and has a
+            ``derivative`` method
+
+        EXAMPLES::
+
+            sage: L.<z> = LazyLaurentSeriesRing(QQ)
+            sage: x = SR.var('x')
+            sage: f(x) = (1 + x)/(1 - x^2)
+            sage: L.taylor(f)
+            1 + z + z^2 + z^3 + z^4 + z^5 + z^6 + O(z^7)
+
+        For inputs as symbolic functions, this uses the generic implementation,
```

Comment:
maybe replace `For inputs as symbolic functions` with `If f is in the symbolic ring`

Reply:
That actually changes things in a very subtle but meaningful way. In particular, polynomials (e.g., `QQ['x,y']`) are also in `SR` but are not symbolic functions. So I think it is better to not change this.